### PR TITLE
[SEDONA-631] Add ST_Expand

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-631. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- Add ST_Expand

## How was this patch tested?

- add new unit tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation.

### CODE REVIEW

File: common/src/main/java/org/apache/sedona/common/utils/BBox.java [hunk 1]
```diff
@@ -38,6 +37,64 @@ public BBox(double startLon, double endLon, double startLat, double endLat) {
...
+    startLon = Double.MAX_VALUE;
+    endLon = -Double.MAX_VALUE;
+    startLat = Double.MAX_VALUE;
+    endLat = -Double.MAX_VALUE;
+    minZ = Double.MAX_VALUE;
+    maxZ = -Double.MAX_VALUE;
+    processEnvelope(geom);
+  }
+
+  private void processEnvelope(Geometry geom) {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/utils/BBox.java:50]: Why is this so complicated? You can see how existing approach read the coordinates https://github.com/apache/sedona/blob/master/common/src/main/java/org/apache/sedona/common/Functions.java#L495

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/sedona/common/Functions.java [hunk 2]
```diff
@@ -90,6 +93,57 @@ public static Geometry boundary(Geometry geometry) {
...
+    double minZ = Double.MAX_VALUE;
+    double maxZ = -Double.MAX_VALUE;
+
+    for (int i = 0; i < coordinates.length; i++) {
+      minX = Math.min(minX, coordinates[i].x);
+      maxX = Math.max(maxX, coordinates[i].x);
+      minY = Math.min(minY, coordinates[i].y);
+      maxY = Math.max(maxY, coordinates[i].y);
+      minZ = Math.min(minZ, coordinates[i].z);
+      maxZ = Math.max(maxZ, coordinates[i].z);
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/Functions.java:123]: When a Geometry does not have Z value, JTS will give NaN (https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Coordinate.html#getZ--). Does the `Math` comparison still work in this case?

File: docs/api/sql/Function.md [hunk 3]
```diff
@@ -1216,6 +1216,38 @@ Output:
 POLYGON ((0 0, 0 3, 1 3, 1 0, 0 0))
 ```
 
+## ST_Expand
+
+Introduction: Returns a geometry expanded from the bounding box of the input. The expansion can be specified in two ways:
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:1221]: Please say this only deal with X, Y, Z value.

If `uniformDelta` is used and the geometry has `Z`, all X, Y, Z will be expanded. If only `deltaX` and `deltaY` are specified, only X and Y will be expanded even if the geometry has `Z`

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/sedona/common/Functions.java (hunk 2 above)
Comment (furqaankhan) [reply common/src/main/java/org/apache/sedona/common/Functions.java:123]: Yeah, it just returns a NaN, and I am skipping the Z values when Geometry doesn't have it. I don't see it being a problem.

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Can you add a test here? https://github.com/furqaankhan/sedona/blob/st-expand/spark/common/src/test/scala/org/apache/sedona/sql/PreserveSRIDSuite.scala

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**